src/App.tsx

```diff
@@ -14,11 +14,154 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  isVisible: boolean;
+  isReversed: boolean;
+  length: number;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    isVisible: false,
+    isReversed: false,
+    length: 1,
+  };
+
+  start = () => {
+    this.setState({ isVisible: true });
+  };
+
+  reverse = () => {
+    this.setState((state) => (
+      { isReversed: !state.isReversed }
+    ));
+  };
+
+  sortByName = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((good1, good2) => (
+        (good1).localeCompare(good2)
+      )),
+    }));
+  };
```

Comment:
maybe store sorting criteria in the state and avoid doing the same thing (sorting) twice?